src/app/components/configuration/configuration.service.js

```diff
@@ -0,0 +1,58 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('gcloudConsole')
+    .service('configuration', Configuration);
+
+  function Configuration($q, GData, firebase) {
+    this.userId = GData.getUser().id;
+    this.$q = $q;
+
+    var url = 'https://phoenix-storage.firebaseio.com/configurations';
+    this.getRef = firebase.getRef(url);
+  }
+
+  Configuration.prototype.getByName = function(name) {
+    var deferred = this.$q.defer();
+
+    this.getRef.then(function(configRef) {
+      configRef.child(name).once('value', function(snap) {
+        var configuration = snap.val();
+
+        if (!configuration) {
+          deferred.reject(new Error('Configuration not found.'));
+          return;
+        }
+
+        deferred.resolve(configuration);
+      });
+    });
+
+    return deferred.promise;
+  };
```

Comment:
I always don't like that `getRef` is a property, as opposed to a function call. The whole thing screams promise noob to me.

Good catch on a better way to wrap this; I didn't even think of that.

RE: event emitter, after using Firebase APIs for so long, I barely consider than an event emitter, and just "how you get values"-- which it is. Firebase is mostly focused on real-time UI/model updates, so the nature of `.on('value')` events fit well. But there isn't a one-off "just get me the value" method, it's just the `once`.